mojo/elements/element.py

```diff
@@ -3,12 +3,32 @@
 from abc import ABC
 from typing import TYPE_CHECKING
 
+import mujoco
+import numpy as np
 from dm_control import mjcf
 
 if TYPE_CHECKING:
     from mojo import Mojo
 
 
+def _is_kinematic(elem: mjcf.Element):
+    if elem.parent is None:
+        # Root of tree
+        return False
+    has_freejoint = hasattr(elem, "freejoint") and elem.freejoint is not None
+    has_joints = hasattr(elem, "joint") and len(elem.joint) > 0
+    return has_freejoint or has_joints or _is_kinematic(elem.parent)
+
+
+def _find_freejoint(elem: mjcf.Element):
+    if elem.parent is None:
+        # Root of tree
+        return None
+    if free_joint := getattr(elem, "freejoint", None):
+        return free_joint
+    return _find_freejoint(elem.parent)
+
+
 class MujocoElement(ABC):
     def __init__(self, mojo: Mojo, mjcf_elem: mjcf.RootElement):
```

Comment:
Could we change typing from `mjcf_elem: mjcf.RootElement` to `mjcf_elem: mjcf.Element`?

It is convenient to create `MujocoElement`s during scene init like this:
`MujocoElement(self._mojo, mjcf_utils.safe_find(model, "body", "body_name"))`

Reply:
Done